pkg/services/ngalert/schedule/compat.go

```diff
@@ -125,3 +126,19 @@ func FromAlertStateToPostableAlerts(firingStates []*state.State, stateManager *s
 	stateManager.Put(sentAlerts)
 	return alerts
 }
+
+// FromAlertsStateToStoppedAlert converts firingStates that have evaluation state eval.Alerting to models.PostableAlert that are accepted by notifiers.
```

Comment:
...or eval.Error or eval.NoData

Is there any reason we couldn't stop the alert regardless of its state? It seems like in the event of an update we'd always want to get rid resources attached to the old alert. Having this behavior be orthogonal and always happen is a bit easier to reason about.

(Unless "Stopped" has a special meaning in the DB and this would cause the non-firing alert would show up in an unexpected place. If this is the case, this is probably fine. It'd be nice if other components were smart enough to avoid such an invalid transition if it were to exist, though.)

Reply:
This package decides whether `State` should be converted to `PostableAlert`. Other functions of this package ignore state Normal and Pending as it does not need an alert to be created in the Alertmanager. The scheduler that gets states from the state manager should not be aware of such peculiarities. 
I forgot to update the comment when I updated this function. Previously, when this PR was created, we handled only Alerting state. After https://github.com/grafana/grafana/pull/40540 and https://github.com/grafana/grafana/pull/41869 we start treating NoData and Error as alerting states.